### PR TITLE
Implement the new Relationship API

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

-   [x] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

-   New route to revoke relationship
    -   `PUT /api/v2/Relationships/{id}/Revoke`
-   Remove relationship changes in favor of new routes

    -   REMOVED `PUT /api/v2/Relationships/{id}/Changes/{changeId}/Reject`
    -   NEW `PUT /api/v2/Relationships/{id}/Reject`
    -   REMOVED `PUT /api/v2/Relationships/{id}/Changes/{changeId}/Accept`
    -   NEW `PUT /api/v2/Relationships/{id}/Accept`

-   update runtime version to `5.0.0-alpha.1`

### CODE REVIEW

**Comment by jkoenig134** [comment]
We need to publish pre-release npm packages for the libs to merge this, right?

File: test/messages.test.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ describe("Messaging", () => {
 });
 
 describe("Message errors", () => {
-    const fakeAddress = "id1PNvUP4jHD74qo6usnWNoaFGFf33MXZi6c";
+    const fakeAddress = "did:e:fakeAddress:dids:0123456789012345678912";
```
Comment (jkoenig134) [inline test/messages.test.ts:96]:
```suggestion
    const fakeAddress = "did:e:a-domain:dids:0000000000000000000000";
```
we should use this everywhere we use a valid but fake address.

File: src/modules/coreHttpApi/openapi.yml
Comment (jkoenig134) [inline src/modules/coreHttpApi/openapi.yml:1]: I'm missing revoke here

Comment (sebbi08) [reply src/modules/coreHttpApi/openapi.yml:1]: This is only "old" functionality updated to V5 the Termination will be handled in another PR

File: src/modules/autoAcceptRelationshipCreationChanges/AutoAcceptRelationshipCreationChangesModule.ts [hunk 2]
```diff
@@ -39,10 +37,11 @@ export default class AutoAcceptRelationshipCreationChangesModule extends Connect
 
     private isIncomingPendingRelationshipCreationChange(event: RelationshipChangedEvent) {
         const data = event.data;
-        if (data.changes.length !== 1) return false;
+        if (data.status !== RelationshipStatus.Pending) return false;
 
-        const creationChange = event.data.changes[0];
-        return creationChange.request.createdBy !== this.currentIdentity && creationChange.status === RelationshipChangeStatus.Pending;
+        const creationChange = event.data.auditLog.find((log) => !log.oldStatus && log.newStatus === RelationshipStatus.Pending);
```
Comment (jkoenig134) [inline src/modules/autoAcceptRelationshipCreationChanges/AutoAcceptRelationshipCreationChangesModule.ts:42]: I think we can massively simplify this.

1. there is only one entry here when the relationship is pending, so you can still look at index 0 (maybe checking for the length and return if its longer than expected)
2. please don't call it creationChange, instead just auditLogEntry
3. you can then just check for the createdBy for the entry on index 0 like you already do

File: packages/sdk/src/types/relationships/ConnectorRelationship.ts [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
+export interface ConnectorRelationshipAuditLogEntry {
+    createdAt: string;
+    createdBy: string;
+    createdByDevice: string;
+    reason: ConnectorRelationshipAuditLogEntryReason;
+    oldStatus?: ConnectorRelationshipStatus;
+    newStatus: ConnectorRelationshipStatus;
+}
+
+export interface ConnectorRelationshipAuditLog extends Array<ConnectorRelationshipAuditLogEntry> {}
```
Comment (jkoenig134) [inline packages/sdk/src/types/relationships/ConnectorRelationship.ts:22]: should be a separate file like it was with the RelationshipChangesBefore

File: src/modules/coreHttpApi/openapi.yml
Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:1]: revoke is not connected to the termination in any way.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:1]: (the revoke I am talking about here)

Comment (sebbi08) [reply src/modules/coreHttpApi/openapi.yml:1]: I cannot find any reference to revoke in the connector.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:1]: I dunno why we had no revoke on relationship changes, but IMO we should provide it now.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@nmshd/connector",
-    "version": "3.11.2",
```
Comment (jkoenig134) [inline package.json:3]: 1. branch says release/v5, please change the branch name
2. as long as the runtime runs under prereleases we have to also publish prereleases here (`4.0.0-alpha.1` see https://github.com/nmshd/runtime/pull/146)

Comment (jkoenig134) [reply package.json:3]: (same for the SDK)

File: packages/sdk/src/types/ConnectorResponse.ts [hunk 5]
```diff
@@ -38,7 +38,9 @@ export class ConnectorResponse<T> {
 
     public get result(): T {
         if (!this.isSuccess) {
-            throw new Error(`Can't get the value of an error Response. Use 'error' instead. Root error: ${this.error.code} - ${this.error.message}`);
+            throw new Error(
+                `Can't get the value of an error Response. Use 'error' instead. Root error: ${this.error.code} - ${this.error.stacktrace?.length ? this.error.stacktrace.join("\n") : this.error.message}`
```
Comment (jkoenig134) [inline packages/sdk/src/types/ConnectorResponse.ts:42]: you hide the message if there is a stack trace

Comment (sebbi08) [reply packages/sdk/src/types/ConnectorResponse.ts:42]: Yea you might think so but JS Error stacktraces always contain the message also a bit weird

Comment (jkoenig134) [reply packages/sdk/src/types/ConnectorResponse.ts:42]: IC

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**